src/Lib.P2P.Tests/SwarmTest.cs

```diff
@@ -0,0 +1,1251 @@
+                Assert.AreEqual(_self.Id, me.Id);
+                Assert.AreEqual(_self.PublicKey, me.PublicKey);
+                Assert.IsNotNull(me.ConnectedAddress);
+
+                // Check disconnect
+                await swarm.DisconnectAsync(peerBAddress);
+                Assert.IsNull(remotePeer.ConnectedAddress);
+                Assert.IsTrue(swarm.KnownPeers.Contains(peerB));
+                Assert.IsTrue(swarmB.KnownPeers.Contains(_self));
+
+                // wait for swarmB to settle
+                endTime = DateTime.Now.AddSeconds(3);
+                while (true)
+                {
+                    if (DateTime.Now > endTime)
+                        Assert.Fail("swarmB did not close connection.");
+                    if (me.ConnectedAddress == null)
+                        break;
+                    await Task.Delay(100);
+                }
+
+                // Reconnect
+                remotePeer = (await swarm.ConnectAsync(peerBAddress)).RemotePeer;
+                Assert.IsNotNull(remotePeer.ConnectedAddress);
+                Assert.AreEqual(peerB.PublicKey, remotePeer.PublicKey);
+                Assert.IsTrue(remotePeer.IsValid());
+                Assert.IsTrue(swarm.KnownPeers.Contains(peerB));
+            }
+            finally
+            {
+                await swarm.StopAsync();
+                await swarmB.StopAsync();
+            }
+        }
+
+        [TestMethod]
+        public async Task RemotePeer_Contains_ConnectedAddress1()
+        {
+            var peerB = new Peer
+            {
+                AgentVersion = "peerB",
+                Id = "QmdpwjdB94eNm2Lcvp9JqoCxswo3AKQqjLuNZyLixmCM1h",
+                PublicKey =
+                    "CAASXjBcMA0GCSqGSIb3DQEBAQUAA0sAMEgCQQDlTSgVLprWaXfmxDr92DJE1FP0wOexhulPqXSTsNh5ot6j+UiuMgwb0shSPKzLx9AuTolCGhnwpTBYHVhFoBErAgMBAAE="
+            };
+            var swarmB = new SwarmService {LocalPeer = peerB};
+            await swarmB.StartAsync();
+            var peerBAddress = await swarmB.StartListeningAsync("/ip4/0.0.0.0/tcp/0");
+
+            var swarm = new SwarmService {LocalPeer = _self};
+            await swarm.StartAsync();
+            try
+            {
+                var connection = await swarm.ConnectAsync(peerBAddress);
+                var remote = connection.RemotePeer;
+                Assert.AreEqual(remote.ConnectedAddress, peerBAddress);
+                CollectionAssert.Contains(remote.Addresses.ToArray(), peerBAddress);
+            }
+            finally
+            {
+                await swarm.StopAsync();
+                await swarmB.StopAsync();
+            }
+        }
+
+        [TestMethod]
+        public async Task RemotePeer_Contains_ConnectedAddress2()
+        {
+            // Only works on Windows because connecting to 127.0.0.100 is allowed
+            // when listening on 0.0.0.0
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows)) return;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)